tempesta_fw/ss_skb.c

```diff
@@ -816,6 +816,50 @@ __skb_fragment(struct sk_buff *skb_head, struct sk_buff *skb, char *pspt,
 			ret = __split_pgfrag(skb_head, skb, i, offset, len, it, fragn);
 			goto done;
 		}
+		/*
+		 * In case another fragment does not exist, remember the
+		 * fragment number, after which you need to insert/delete data.
+		 */
+		if (offset == d_size)
+			j = i;
+	}
+
+	if (j >= 0) {
+		skb_frag_t *frag = &si->frags[j + 1];
```

Comment:
We can get here with j == si->nr_frags - 1, so this could be an out of bound access here and in all the statements till the end of the if statement. Here we can check the `j` value and increment it to deal with `j` till the end of the if statement.